src/baldr/graphreader.cc

```diff
@@ -880,5 +880,53 @@ int GraphReader::GetTimezone(const baldr::GraphId& node, const GraphTile*& tile)
   return (tile == nullptr) ? 0 : tile->node(node)->timezone();
 }
 
+IncidentResult GraphReader::GetIncidents(const GraphId& edge_id, const GraphTile*& tile) {
+  // if we are not doing this for any reason then bail
+  std::shared_ptr<valhalla::incidents::IncidentsTile> itile;
+  if (!simulate_incidents_ || !GetGraphTile(edge_id, tile) ||
+      !tile->trafficspeed(tile->directededge(edge_id)).has_incidents ||
+      !(itile = GetIncidentTile(edge_id))) {
+    return {};
+  }
+
+  // get the range of incidents we care about and hand it back, equal_range has no lambda option
+  auto begin =
+      std::partition_point(itile->incident_locations().begin(), itile->incident_locations().end(),
+                           [&edge_id](const valhalla::incidents::IncidentLocation& candidate) {
+                             // first one that is >= the id we want
+                             bool is_found = candidate.edge_index() < edge_id.id();
+                             return is_found;
+                           });
+  auto end = std::partition_point(begin, itile->incident_locations().end(),
+                                  [&edge_id](const valhalla::incidents::IncidentLocation& candidate) {
+                                    bool is_found =
+                                        candidate.edge_index() <=
+                                        edge_id.id(); // first one that is > the id we want
+                                    return is_found;
+                                  });
+
+  int begin_index = begin - itile->incident_locations().begin();
+  int end_index = end - itile->incident_locations().begin();
+
+  if (begin_index >= itile->incident_locations_size()) {
+    // No incidents
+    return {nullptr, 0, 0};
+  } else {
+    return {itile, begin_index, end_index};
+  }
```

Comment:
no need to add an `if` here, both indices will be `itile->incident_locations_size()` when the edge index is not found